sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/PlanTest.scala

```diff
@@ -198,4 +201,23 @@ trait PlanTestBase extends PredicateHelper with SQLHelper { self: Suite =>
         plan1 == plan2
     }
   }
+
+  def assertEqualPlansForJoinReorder(
```

Comment:
Can we create a base test suite for join reorder?

Reply:
yes I also think that's better. Do you think it's necessary to create a separate folder called "joinReorder" under the optimizer package?

Reply:
@cloud-fan Updated. I add a new folder "joinReorder" under the "optimizer" test package, and put 3 suites and the new base test in it.

Reply:
Ur, @cloud-fan and @wzhfy , why a new test trait is better than this? The function looks useful in general.